osbs/utils.py

```diff
@@ -256,7 +260,11 @@ def clone_git_repo(git_url, target_dir=None, commit=None, retry_times=GIT_MAX_RE
             # we are using check_output, even though we aren't using
             # the return value, but we will get 'output' in exception
             subprocess.check_output(cmd, stderr=subprocess.STDOUT)
-            repo_commit, repo_depth = reset_git_repo(target_dir, commit, depth)
+            try:
+                repo_commit, repo_depth = reset_git_repo(target_dir, commit, depth)
+            except OsbsException as exc:
+                raise OsbsCommitNotFound("Commit %s is not reachable in branch %s, reason: %s"
+                                         % (commit, branch, exc.message))
```

Comment:
Please don't use `exc.message`. It's deprecated. Use just plain exception object it will be converted to string when you use `%s` formatting.

Also I'd prefer `.format()` formatting instead of `%`. It's never and better for future compatibility with F-string in py36+